planner_cspace/src/planner_3d.cpp

```diff
@@ -1623,38 +1623,59 @@ class Planner3dNode
   }
 
 protected:
-  bool makePlan(const geometry_msgs::Pose& gs, const geometry_msgs::Pose& ge,
-                nav_msgs::Path& path, bool hyst)
+  void publishStartAndGoalMarkers(const Astar::Vec& start_grid, Astar::Vec& end_grid)
   {
-    Astar::Vec e;
-    grid_metric_converter::metric2Grid(
-        map_info_, e[0], e[1], e[2],
-        ge.position.x, ge.position.y, tf2::getYaw(ge.orientation));
-    e.cycleUnsigned(map_info_.angle);
+    geometry_msgs::PoseStamped p;
+    p.header = map_header_;
+    float x, y, yaw;
+    grid_metric_converter::grid2Metric(map_info_, end_grid[0], end_grid[1], end_grid[2], x, y, yaw);
+    p.pose.orientation = tf2::toMsg(tf2::Quaternion(tf2::Vector3(0.0, 0.0, 1.0), yaw));
+    p.pose.position.x = x;
+    p.pose.position.y = y;
+    pub_end_.publish(p);
+    grid_metric_converter::grid2Metric(map_info_, start_grid[0], start_grid[1], start_grid[2], x, y, yaw);
+    p.pose.orientation = tf2::toMsg(tf2::Quaternion(tf2::Vector3(0.0, 0.0, 1.0), yaw));
+    p.pose.position.x = x;
+    p.pose.position.y = y;
+    pub_start_.publish(p);
+  }
 
+  enum class StartPoseStatus
+  {
+    START_OCCUPIED,
+    FINISHING,
+    NORMAL,
+  };
+  StartPoseStatus buildStartPoses(const geometry_msgs::Pose& start_metric, const geometry_msgs::Pose& end_metric,
+                                  std::vector<Astar::VecWithCost>& result_start_poses) const
+  {
     Astar::Vecf sf;
     grid_metric_converter::metric2Grid(
         map_info_, sf[0], sf[1], sf[2],
-        gs.position.x, gs.position.y, tf2::getYaw(gs.orientation));
-    Astar::Vec s(static_cast<int>(std::floor(sf[0])), static_cast<int>(std::floor(sf[1])), std::lround(sf[2]));
-    s.cycleUnsigned(map_info_.angle);
-    if (!cm_.validate(s, range_))
+        start_metric.position.x, start_metric.position.y, tf2::getYaw(start_metric.orientation));
+    Astar::Vec start_grid(static_cast<int>(std::floor(sf[0])), static_cast<int>(std::floor(sf[1])), std::lround(sf[2]));
+    start_grid.cycleUnsigned(map_info_.angle);
+
+    Astar::Vec end_grid;
+    grid_metric_converter::metric2Grid(
+        map_info_, end_grid[0], end_grid[1], end_grid[2],
+        end_metric.position.x, end_metric.position.y, tf2::getYaw(end_metric.orientation));
+    end_grid.cycleUnsigned(map_info_.angle);
+
+    if (!cm_.validate(start_grid, range_))
     {
       ROS_ERROR("You are on the edge of the world.");
```

Comment:
I think logs in `buildStartPoses` shouldn't be shown on GetPlan service call

Reply:
`cbMakePlan()` does not call this function as it uses the start pose designated in `GetPlan::Request` without anti-alias or relocations.